strawberryfields/io.py

```diff
@@ -0,0 +1,214 @@
+
+        if "Measure" in op["op"]:
+            # special case to take into account 'select' keyword argument
+            if cmd.op.select is not None:
+                op["kwargs"]["select"] = cmd.op.select
+
+            if cmd.op.p:
+                # argument is quadrature phase
+                op["kwargs"]["phi"] = cmd.op.p[0].x
+
+        else:
+            if cmd.op.p is not None:
+                for a in cmd.op.p:
+                    # check if reg ref transform
+                    if isinstance(a.x, ops.RegRefTransform):
+                        if a.x.func_str is not None:
+                            # Note: not currently guaranteed to work,
+                            # as the RegRefTransform string cannot be checked
+                            # to determine if it is a valid function for serialization!
+                            #
+                            # Possible solutions:
+                            #
+                            #   * Use SymPy to represent functions, as
+                            #     SymPy provides methods for converting to
+                            #     Python functions as well as serialization
+                            #
+                            #   * Don't allow classical processing of measurements
+                            #     on remote backends
+                            #
+                            op["args"].append(a.x.func_str)
+                        else:
+                            raise ValueError(
+                                "The RegRefTransform in operation {} "
+                                "is not supported by Blackbird.".format(cmd.op)
+                            )
+                    else:
+                        op["args"].append(a.x)
+
+        bb._operations.append(op)
+
+    return bb
+
+
+def to_program(bb):
+    """Convert a Blackbird Program to a Strawberry Fields Program.
+
+    Args:
+        bb (blackbird.BlackbirdProgram): the input Blackbird program object
+
+    Returns:
+        Program: corresponding SF program
+    """
+    # create a SF program
+    if not bb.modes:
+        # we can't return an empty program, since we don't know how many modes
+        # to initialize the Program object with.
+        raise ValueError("Blackbird program contains no quantum operations!")
+
+    prog = Program(max(bb.modes)+1, name=bb.name)
+
+    # append the quantum operations
+    with prog.context as q:
+        for op in bb.operations:
+            try:
+                if 'args' in op:
+                    getattr(ops, op["op"])(*op["args"], **op["kwargs"]) | [q[i] for i in op["modes"]] #pylint:disable=expression-not-assigned
+                else:
+                    getattr(ops, op["op"]) | [q[i] for i in op["modes"]] #pylint:disable=expression-not-assigned
+            except AttributeError:
+                raise NameError("Quantum operation {} not defined!".format(op["op"]))
+
+    # compile the program if a target exists
```

Comment:
Do we want to do an implicit compilation of a program? Or do we want to just convert from bb program to sf program, and leave it up to user to compile?

Reply:
I would argue for implicit conversion, personally.

Reply:
I'm not convinced :thinking:

Reply:
Based on further discussion, it makes more sense for the basic load methods to just do deserialization, and no validation. Any target that appears in the blackbird program can be added as an attribute to the corresponding `Program` instance